RONDB-205: Add options --mysqld-instrumentation and --extra-packages

svenssonaxel opened the pull request.

svenssonaxel committed:
RONDB-205: Add options --mysqld-instrumentation and --extra-packages

olapiv reviewed (commented):
Looks good. Just wondering whether the Docker image tag should be changed accordingly if we install the extra packages? I don't want to rebuild everything if I try out this feature.

olapiv commented on build_run_docker.sh:
```diff
@@ -219,17 +240,29 @@ fi
+# context.
+MYSQLD_INSTRUMENTATION_INDICATOR=
+if [ -n "$MYSQLD_INSTRUMENTATION" ]; then
+    MYSQLD_INSTRUMENTATION_INDICATOR="$(echo "$MYSQLD_INSTRUMENTATION" | sha256sum | sed -r 's/^(.{8}).*$/_mi-\1/')"
```
I think 3 characters is already enough? 256x256x256=16m combinations

svenssonaxel replied:
The hash sum is in hexadecimal, so if you want 24 bits of information then that's 6 characters.

olapiv replied:
Ah right, okay. 16^4 is okay (65k)? I don't expect ppl to try huge amounts of different combinations without deleting the folders in-between.

olapiv replied:
(Sorry, I just get annoyed by the long folder names in my IDE 😅)

svenssonaxel replied:
Agreed, with 16⁴ hashes we'd expect a collision after about √ (16⁴*π/2)≈320 evaluations, and if that many are used one's likely to try to parallelize anyway, and then we have a clash in container names instead.

svenssonaxel replied:
Fixed in `ee54619`

svenssonaxel commented:
> Looks good. Just wondering whether the Docker image tag should be changed accordingly if we install the extra packages? I don't want to rebuild everything if I try out this feature.

Fair point. If the tag is overwritten and you clear the cache you'd have to rebuild, so it's probably better to make unique tags. I'll push a fix tomorrow.

olapiv commented:
Just tested it and for me the `sleep 25` for the API node did not suffice. It takes more like 120s with instrumentation. But I've exchanged the ndb_waiter for a Docker MySQLd healthcheck in the rest-api branch that I'm working on, which will work a lot more accurately anyways. The ndb_waiter returns `STARTED` if the cluster has started - that does not mean that the MySQL server is ready yet.

svenssonaxel commented:
> I've exchanged the ndb_waiter for a Docker MySQLd healthcheck in the rest-api branch that I'm working on

Perhaps it would make sense to wait with this PR until that is merged?

olapiv commented:
Yeah maybe. Could still take a couple of weeks, but then we're sure this works without hacking the script.

svenssonaxel commented:
> Yeah maybe. Could still take a couple of weeks, but then we're sure this works without hacking the script.

I don't think we're in a hurry with this one.

svenssonaxel committed:
RONDB-205: Fixes

- Use only 4 hexadecimal digits to identify instrumentation command line.
- Use only 4 hexadecimal digits to identify extra packages.
- Let tag name depend on instrumentation command line and extra packages.

svenssonaxel commented:
> Docker image tag should be changed accordingly

Fixed in `ee54619`.

svenssonaxel committed:
Merge branch 'main' into RONDB-205-instrumentation

olapiv commented:
Needs https://github.com/logicalclocks/rondb-docker/pull/16

smkniazi commented:
Closing as it has no activity for 2 years
